### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by FrankMugagga**
### Activities
Created a table named vets
Established many-to-many relationship between the tables species and vets and created table specializations
Established a many-to-many relationship between the tables animals and vets and created a table visits
Inserted corresponding data in the created tables
Did queries that required joining the tables
Updated the REAME.md file

### CODE REVIEW

**Review by Kamba56 — COMMENTED** [review]
Hi @FrankMugagga

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly remove the Live demo section from your readme as it does not apply to this project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: queries.sql [hunk 1]
```diff
@@ -39,6 +39,31 @@ SELECT AVG(weight_kg) FROM animals;
...
+SELECT animals.name AS animal_name, vets.name, MIN(visits.date_of_visit) FROM VISITS JOIN animals ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE vets.name = 'Maisy Smith' group by animals.name, vets.name, visits.date_of_visit limit 1;
+SELECT * FROM visits JOIN animals ON visits.animals_id = animals.id JOIN vets ON vets.id = visits.vets_id ORDER BY visits.date_of_visit desc limit 1;
+SELECT COUNT(*) FROM visits JOIN (SELECT vets.id FROM vets FULL JOIN specializations ON vets.id = specializations.vets_id FULL JOIN species ON species.id = specializations.species_id WHERE specializations.species_id IS NULL) AS
+ vet ON vet.id = visits.vets_id;
+ SELECT s.name AS species_name, COUNT(vis.date_of_visit) AS total_visits FROM visits AS vis JOIN animals AS 
+a ON vis.animals_id = a.id JOIN species AS s ON a.species_id = s.id JOIN vets AS v ON vis.vets_id = v.id WHERE v.name = 
+ 'Maisy Smith' GROUP BY s.name ORDER BY total_visits DESC LIMIT 1;
```
Comment (Kamba56) [inline queries.sql]: - [ ] Kindly add screenshots of the results you got after running all queries to your PR. The screenshots will determine if you successfully ran all the queries and if the queries are correct.

File: queries.sql [hunk 2]
```diff
@@ -39,6 +39,31 @@ SELECT AVG(weight_kg) FROM animals;
...
+SELECT vets.name, species.name, specializations.species_id FROM SPECIALIZATIONS FULL JOIN vets ON specializations.vets_id = vets.id FULL JOIN species ON species.id = specializations. species_id GROUP BY specializati
+ons.species_id, vets.name, species.name;
+SELECT animals.name AS animals_that_visited, visits.date_of_visit, vets.name FROM VISITS JOIN ANIMALS ON animals.id = visits.animals_id JOIN vets ON vets.id = visits.vets_id WHERE vets.id = 3 AND date_of_visit BETWEEN '2020-04-1' AND '2020-08-30' GROUP BY animals.name, vets.name, visits.date_of_visit;
+SELECT animals.name, COUNT(visits.animals_id) AS Animal_visits FROM visits JOIN animals ON animals.id = animals_id JOIN vets ON vets.id = visits.vets_id GROUP BY animals.name ORDER BY Animal_visits desc;
+SELECT animals.name, COUNT(visits.animals_id) AS Animal_visits FROM visits JOIN animals ON animals.id = animals_id JOIN vets ON vets.id = visits.vets_id GROUP BY animals.name ORDER BY Animal_visits;
+vet_clinic=# SELECT animals.name, COUNT(visits.animals_id) AS Animal_visits FROM visits JOIN animals ON animals.id = animals_id JOIN vets ON vets.id = visits.vets_id GROUP BY animals.name ORDER BY Animal_visits desc limit 2;
```
Comment (Kamba56) [inline queries.sql]: - [ ] Remove all commented pieces of code from your documents to maintain a clean document.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @FrankMugagga, 

Good job so far!!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏!👏👏👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Great job adding a relationship between vets and species using the specialization table.!👏👏👏
- [x] Great job updating the tables. 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by arslanbisharat** [comment]
Hi @FrankMugagga,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**